src/go/k8s/apis/redpanda/v1alpha1/console_common.go

```diff
@@ -0,0 +1,50 @@
+package v1alpha1
+
+import (
+	"github.com/redpanda-data/console/backend/pkg/connect"
+	"github.com/redpanda-data/console/backend/pkg/filesystem"
+	"github.com/redpanda-data/console/backend/pkg/git"
+	"github.com/redpanda-data/console/backend/pkg/msgpack"
+	"github.com/redpanda-data/console/backend/pkg/proto"
+)
+
+// Alias types are defined here because of controller-gen "make generate" does not generate DeepCopyInto function on these types.
+// REF https://github.com/kubernetes-sigs/controller-tools/issues/583
+
+type (
+	// ConnectConfig is an alias to connect.Config
+	ConnectConfig connect.Config
+
+	// MsgpackConfig is an alias to msgpack.Config
+	MsgpackConfig msgpack.Config
+
+	// GitConfig is an alias to git.Config
+	GitConfig git.Config
+
+	// FilesystemConfig is an alias to filesystem.Config
+	FilesystemConfig filesystem.Config
+)
+
+// ConsoleConfig is a redefinition of console.Config but struct fields defined locally
+type ConsoleConfig struct {
+	TopicDocumentation ConfigTopicDocumentation `json:"topicDocumentation"`
+}
+
+// ConfigTopicDocumentation is a redefinition of console.ConfigTopicDocumentation but struct fields defined locally
+type ConfigTopicDocumentation struct {
+	Enabled bool      `json:"enabled"`
+	Git     GitConfig `json:"git"`
+}
+
+// ProtoConfig is a redefinition of console.ProtoConfig but struct fields defined locally
+type ProtoConfig struct {
+	Enabled bool `json:"enabled"`
+
+	// The required proto definitions can be provided via SchemaRegistry, Git or Filesystem
+	SchemaRegistry proto.SchemaRegistryConfig `json:"schemaRegistry"`
+	Git            GitConfig                  `json:"git"`
+	FileSystem     FilesystemConfig           `json:"fileSystem"`
```

Comment:
How does it work in a k8s environment? Are users allowed to mount volumes?

Reply:
Right now you cannot specify this in Console CRD but technically users can update volumeMounts of the deployment. I think we are going with representing as much of the Console config as we can for flexibility. https://github.com/redpanda-data/console/blob/master/docs/config/console.yaml#L85